validator/project_validator.go

```diff
@@ -2347,6 +2348,32 @@ func checkTasks(project *model.Project) ValidationErrors {
 	return errs
 }
 
+// checkTaskUsage returns a notice for each task that is defined but unused by any (un-disabled) variant.
+// TODO: upgrade to a warning in DEVPROD-8154
+func checkTaskUsage(project *model.Project) ValidationErrors {
+	errs := ValidationErrors{}
+	usedTasks := map[string]bool{}
```

Comment:
nit: The value is never used, and could cause a bit of confusion with the `ok` check below - that's checking that the key exists but could be mistaken for checking if the value is true since either one would return boolean. Could we consider doing that thing in Go where we use `struct{}` as the map value to indicate that the map is only being used as a set, and the value is irrelevant? There's an example [here](https://github.com/evergreen-ci/evergreen/blob/30fb406e455a38d4769421cfa103bb97a4d65803/model/generate.go#L224).

Reply:
Haha i think you and i have this conversation like once a year 😂 I think that makes sense but also I might just change the if statement to use the value rather than the ok, since that works just as well here, and I like the cleanness of boolean assignments. Feel free to push back if you want us to standardize this.

Reply:
Eh, I don't care that much, either way works. I'm fine with us having slightly different preferences for making sets since either way is a little weird.

And yeah, maybe I'll forget about this and we'll be doing this over again in 2025 😂